lib/ContextModule.js

```diff
@@ -307,6 +329,9 @@ webpackEmptyAsyncContext.id = ${JSON.stringify(id)};`;
 	}
 
 	getSourceString(asyncMode, outputOptions, requestShortener) {
+		if(asyncMode === "weak-context") {
+			return this.getSyncSource(this.weakDependencies, this.id);
```

Comment:
create a new method `getWeakSyncSource`.

You can copy the `getSyncSource` method, but add a check on `require` like this

``` js
function webpackContext(req) {
  var id = webpackContextResolve(req);
  if(!__webpack_require__.m[id])
    throw new Error("Module '" + id + "' is not available (weak dependency)");
  return __webpack_require__(id);
};
```

And do the same of `getWeakEagerSource` with `asyncMode === "eager-weak"`. The difference to `getWeakSyncSource` is that it returns a Promise so it can be used for 
``import(/* webpackMode: "eager-weak" */`./templates/${expr}.js`)``